fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneQueryMultiFieldSearchClause.java

```diff
@@ -64,7 +70,23 @@ public Query bind(@Nonnull FDBRecordStoreBase<?> store, @Nonnull Index index, @N
         final LuceneAnalyzerCombinationProvider analyzerSelector = LuceneAnalyzerRegistryImpl.instance().getLuceneAnalyzerCombinationProvider(index, LuceneAnalyzerType.FULL_TEXT);
         final String[] fieldNames = LuceneScanParameters.indexTextFields(index, store.getRecordMetaData()).toArray(new String[0]);
         final String searchString = isParameter ? (String)context.getBinding(search) : search;
-        final QueryParser parser = new LuceneOptimizedMultiFieldQueryParser(fieldNames, analyzerSelector.provideQueryAnalyzer(searchString).getAnalyzer());
+        final LuceneOptimizedMultiFieldQueryParser parser = new LuceneOptimizedMultiFieldQueryParser(fieldNames, analyzerSelector.provideQueryAnalyzer(searchString).getAnalyzer());
+        Map<String, PointsConfig> pointsConfigMap = new HashMap<>();
+        final Collection<RecordType> recordTypes = store.getRecordMetaData().recordTypesForIndex(index);
+        for (RecordType type : recordTypes) {
+            LuceneIndexExpressions.getDocumentFieldDerivations(index.getRootExpression(), type.getDescriptor()).forEach((key, value) -> {
+                PointsConfig valueConfig = value.getPointsConfig();
+                if (valueConfig != null) {
+                    PointsConfig oldConfig = pointsConfigMap.get(key);
+                    if (oldConfig != null && !oldConfig.equals(valueConfig)) {
```

Comment:
This stores the new config in the equals case, which is, of course, equivalent, but maybe a tiny bit more work.